kbase-extension/static/kbase/js/widgets/narrative_core/jobCommChannel.js

```diff
@@ -221,6 +254,113 @@ define([
                 case 'new_job':
                     Jupyter.notebook.save_checkpoint();
                     break;
+
+                // CELL messages
+                case 'run_status':
+                    // Send job status notifications on the default channel,
+                    // with a key on the message type and the job id, sending
+                    // a copy of the original message.
+                    // This allows widgets which are interested in the job
+                    // to subscribe to just that job, and nothing else.
+                    // If there is a need for a generic broadcast message, we
+                    // can either send a second message or implement key
+                    // filtering.
+                    this.sendBusMessage(CELL, msgData.cell_id, 'run-status', msgData);
+                    break;
+
+                case 'result':
+                    this.sendBusMessage(CELL, msgData.address.cell_id, 'result', msgData);
+                    break;
+
+                // JOB messages
+                case 'job_canceled':
+                    this.sendBusMessage(JOB, msgData.job_id, 'job-canceled', {
+                        jobId: msgData.job_id,
+                        via: 'job_canceled',
+                    });
+                    break;
+
+                case 'job_comm_error':
+                    if (msgData) {
+                        jobId = msgData.job_id;
+                        switch (msgData.source) {
+                            case 'cancel_job':
+                                this.sendBusMessage(JOB, jobId, 'job-cancel-error', {
+                                    jobId,
+                                    message: msgData.message,
+                                });
+                                this.sendBusMessage(JOB, jobId, 'job-canceled', {
+                                    jobId,
+                                    error: msgData.message,
+                                });
```

Comment:
Doubling up these messages for now -- the goal is to remove the `job-cancel-error` channel and just use the `job-canceled` channel with an altered data structure. Since this PR doesn't contain any of the listeners to the `job-cancel-error` channel, that message cannot yet be removed. (Will be changing the listeners in an upcoming PR)